pd/src/proxy/tendermint.rs

```diff
@@ -0,0 +1,97 @@
+use penumbra_proto::tendermint_proxy::tendermint_query_server::TendermintQuery;
+use penumbra_proto::tendermint_proxy::{
+    AbciQueryRequest, AbciQueryResponse, GetBlockByHeightRequest, GetBlockByHeightResponse,
+    GetLatestBlockRequest, GetLatestBlockResponse, GetLatestValidatorSetRequest,
+    GetLatestValidatorSetResponse, GetNodeInfoRequest, GetNodeInfoResponse, GetSyncingRequest,
+    GetSyncingResponse, GetValidatorSetByHeightRequest, GetValidatorSetByHeightResponse,
+};
+use tendermint_rpc::Client;
+use tonic;
+
+use tendermint_rpc;
+
+#[derive(Clone)]
+pub struct TendermintProxyStub {
+    pub tendermint_instance: String,
+}
+
+impl Default for TendermintProxyStub {
+    fn default() -> Self {
+        Self {
+            tendermint_instance: "127.0.0.1:26656".to_string(),
+        }
+    }
+}
+
+#[tonic::async_trait]
+impl TendermintQuery for TendermintProxyStub {
+    async fn get_node_info(
+        &self,
+        request: tonic::Request<GetNodeInfoRequest>,
+    ) -> Result<tonic::Response<GetNodeInfoResponse>, tonic::Status> {
+        let mut client =
+            tendermint_rpc::HttpClient::new(self.tendermint_instance.as_str()).unwrap();
+        let resp = client.status().await.unwrap();
+
+        // Find a way to get from RPC types to proto types.
+        // Enquiring with tendermint-rs devs, there was no clean way to do this in the code
+        // I have reviewed. Might be necessary to either:
+        //      - implement `TryFrom` / `TryInto` upstream
+        //      - get the upstream repo to consolidate types
+
+        // Ok(tonic::Response::new(resp.into()));
+        unimplemented!()
+    }
+
+    #[doc = " GetSyncing queries node syncing."]
```

Comment:
Where'd this `doc` attribute come from? Is this pulling comments from the protos somehow?

Reply:
They're artifacts from the tonic codegen, I'll replace them with more descriptive doc comments!